pkg/fission-cli/cmd/spec/validate.go

```diff
@@ -281,3 +339,59 @@ func ReadSpecs(specDir, specIgnore string) (*FissionResources, error) {
 
 	return &fr, nil
 }
+
+func index(slice []string, item string) int {
+	for i := range slice {
+		if slice[i] == item {
+			return i
+		}
+	}
+	return -1
+}
+
+// getCommitLabel returns the value of the 'commit' label
+func getCommitLabel(path, gitRepoRootPath, commitID string, status *git.Status, repo *git.Repository) string {
+
+	gitRepoDirName := filepath.Base(gitRepoRootPath)
+	splitPathList := strings.Split(path, string(os.PathSeparator))
```

Comment:
Please see if we can do this part without splitting the string. Maybe use regex. index function seems unnecessary to me. 
Also, we are not handling the -1 return case.

Reply:
Here I have the absolute path of the files in the specdir and I have the absolute path of the git repository, where spec directory can be the root directory of the git repo or it might be nested inside the git repo tree. So here I am removing the absolute path of the git repo from the absolute specdir file path to get the relative file apth inside the git repository. So this has to exist and the -1 case has been handled. The index function has been removed.